pkg/operator/controller/machineconfig/nodes.go

```diff
@@ -93,27 +93,28 @@ func (r *MachineConfigReconciler) ensureReqNodeLabelExists(ctx context.Context)
 	if err := r.listWorkerNodes(ctx, nodeList); err != nil {
 		return updNodeCount, err
 	}
-
 	for i, node := range nodeList.Items {
-		if _, exist := node.Labels[NodeObservabilityNodeRoleLabelName]; exist {
-			continue
-		}
+		for nodeSelectorKey, nodeSelectorValue := range r.CtrlConfig.Spec.NodeSelector {
+			if _, exist := node.Labels[nodeSelectorKey]; exist {
+				continue
+			}
 
-		patch, _ := newPatch(add,
-			ResourceLabelsPath,
-			map[string]interface{}{
-				NodeObservabilityNodeRoleLabelName: Empty,
-			})
-		if err := r.ClientPatch(ctx, &nodeList.Items[i], client.RawPatch(types.JSONPatchType, patch)); err != nil {
-			return updNodeCount, err
-		}
-		r.Log.V(1).Info("Successfully added label", "Node", node.Name, "Label", NodeObservabilityNodeRoleLabelName)
+			patch, _ := newPatch(add,
+				ResourceLabelsPath,
+				map[string]interface{}{
+					nodeSelectorKey: nodeSelectorValue,
+				})
+			if err := r.ClientPatch(ctx, &nodeList.Items[i], client.RawPatch(types.JSONPatchType, patch)); err != nil {
+				return updNodeCount, err
+			}
```

Comment:
Hi @ayesha54 
I'm not sure if we should add the nodeSelector to all worker nodes: the admin must have already set the right selectors on those nodes he/she likes to perform the pprof on. 
So, instead, I think we should  use it in order to identify nodes to include in the machineConfigPool.
WDYT?

Reply:
Hi @ayesha54 
Methods used for ensuring and removing the labels will not be required anymore, since existing node role label "worker" will be used for selecting the nodes.

Reply:
As we discussed with @ayesha54, we'll keep the current logic of adding the custom nodeobservability role label. The nodeselector passed to `NobMC` has to be used in `List` nodes API call to filter the nodes which will be added with the custom label, not for the labeling.